docs/lib/search.tsx

```diff
@@ -0,0 +1,92 @@
+import { useEffect, useState } from "react";
+import type {
+  PagefindSearchFragment,
+  PagefindSearchResult,
+  PagefindSearchResults,
+} from "./search-types";
+import { mockSearchData } from "./mock-search-data";
+
+export const ignoredRoutes = ["/blog", "/terms", "/privacy", "/confirm"];
+
+export const downrankedRoutes = [
+  "/repo/docs/acknowledgements",
+  // Deprecations
+  "/repo/docs/core-concepts/scopes",
+];
+
+declare global {
+  interface Window {
+    pagefind?: {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any -- Not sure where this type comes from.
+      search: (...args: any[]) => Promise<PagefindSearchResults>;
+    };
+  }
+}
+
+export const usePageFindSearch = () => {
+  useEffect(() => {
+    async function loadPagefind() {
+      if (typeof window.pagefind === "undefined") {
+        try {
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment -- Not sure where to get this type.
+          window.pagefind = await import(
+            // @ts-expect-error -- Generated at buildtime
+            // eslint-disable-next-line import/no-unresolved -- Generated at buildtime
+            /* webpackIgnore: true */ "./pagefind/pagefind.js"
+          );
+        } catch (e) {
+          if (process.env.NODE_ENV === "development") {
+            window.pagefind = {
+              search: () =>
+                new Promise((resolve) => {
+                  resolve({
+                    results: mockSearchData,
+                  } as unknown as PagefindSearchResults);
+                }),
```

Comment:
Re: https://github.com/vercel/turbo/pull/6878/files#r1440465581

Hacking in some results to work with for the sake of a faster development workflow.

Reply:
If there are strong opinions, we can remove this code and have a slower feedback loop on the rare occasions that this search needs to be worked on again.